### PR TITLE
server version test migrated

### PR DESCRIPTION
**Pull request by Kinju011**

### CODE REVIEW

File: spec/neo4j/driver/util/server_version_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+RSpec.describe Neo4j::Driver::Internal::Util::ServerVersion do
+  context 'when', RUBY_PLATFORM: 'ruby' do
```
Comment (klobuczek) [inline spec/neo4j/driver/util/server_version_spec.rb:2]: have you verified that this actually works and the specs do not run on jruby?

File: spec/neo4j/driver/util/server_version_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      expect(Neo4j::Driver::Internal::Util::ServerVersion::V4_1_0).to eq(Neo4j::Driver::Internal::Util::ServerVersion.from_bolt_protocol_version(Neo4j::Driver::Internal::Messaging::V41::BoltProtocolV41::VERSION))
+      expect(Neo4j::Driver::Internal::Util::ServerVersion::V4_2_0).to eq(Neo4j::Driver::Internal::Util::ServerVersion.from_bolt_protocol_version(Neo4j::Driver::Internal::Messaging::V42::BoltProtocolV42::VERSION))
+      expect(Neo4j::Driver::Internal::Util::ServerVersion::V4_3_0).to eq(Neo4j::Driver::Internal::Util::ServerVersion.from_bolt_protocol_version(Neo4j::Driver::Internal::Messaging::V43::BoltProtocolV43::VERSION))
+      expect(Neo4j::Driver::Internal::Util::ServerVersion::V4_4_0).to eq(Neo4j::Driver::Internal::Util::ServerVersion.from_bolt_protocol_version(Neo4j::Driver::Internal::Messaging::V44::BoltProtocolV44::VERSION))
+      expect(Neo4j::Driver::Internal::Util::ServerVersion::V_IN_DEV).to eq(Neo4j::Driver::Internal::Util::ServerVersion.from_bolt_protocol_version(Neo4j::Driver::Internal::Messaging::BoltProtocolVersion.new(java.lang.Integer::MAX_VALUE, java.lang.Integer::MAX_VALUE)))
```
Comment (klobuczek) [inline spec/neo4j/driver/util/server_version_spec.rb:37]: there is no java in ruby
I think I actually mapped those MAX_VALUE to nil

**Review by klobuczek — CHANGES_REQUESTED** [review]

File: spec/neo4j/driver/util/server_version_spec.rb (hunk 2 above)
Comment (Kinju011) [reply spec/neo4j/driver/util/server_version_spec.rb:37]: okay, there is a difference `(MAX_INTEGER = 2 ^ 31 - 1)` in the pending directory `server_version` file while `nil` in `removing_java` branch, but will change it to nil

File: spec/neo4j/driver/util/server_version_spec.rb (hunk 1 above)
Comment (Kinju011) [reply spec/neo4j/driver/util/server_version_spec.rb:2]: I have tested it against java and working fine but will verify again and update it here.

File: spec/neo4j/driver/util/server_version_spec.rb (hunk 2 above)
Comment (klobuczek) [reply spec/neo4j/driver/util/server_version_spec.rb]: yes, I see that you did not implement it against removing_java. Only the classes in `ruby` subdirectory are relevant. The class has a different constructor there, please adjust.

File: spec/neo4j/driver/util/server_version_spec.rb (hunk 1 above)
Comment (klobuczek) [reply spec/neo4j/driver/util/server_version_spec.rb:2]: Working should mean the spec is skipped on jruby.

File: spec/neo4j/driver/util/server_version_spec.rb (hunk 2 above)
Comment (Kinju011) [reply spec/neo4j/driver/util/server_version_spec.rb]: yes, sure

**Pull request merged**